### PR TITLE
fix(dashboards) Enable dashboards for self-hosted

### LINKED ISSUE #25223: No Dashboard on fresh on-premise install
## Important Details

How are you running Sentry?

On-Premise w/ Docker

Sentry 21.4.0.dev053c8331
Docker version 20.10.6, build 370c289
Ubuntu 20.04 fully updated

## Description

Fresh install w/ minimal config changes in sentry directory (system.secret-key, and uncomment items for NGINX https reverse proxy).  Everything seems to be functioning fine except the Dashboard is empty.  After a (very) brief spinner stating “Loading data for your organization” the resulting page is empty except the footer info (version, api, doc, etc links) 

Dashboard page has no console errors and all requests made are 200's. 

Nothing stands out to me in the logs, although with 28 containers I could easily be missing something ;)  No warnings in the admin section.

sentry-dash

## Steps to Reproduce

1. Cloned Sentry on-premise repo
2. Minimal config changes (system.secret-key, and uncomment items for NGINX https reverse proxy)
3. install.sh
4. docker-compose up -d
5. Complete web based initial config

After initial attempt resulted in no data, I did the following:

1. docker-compose down
2. removed *all* remaining Docker volumes and images
3. re-ran install.sh
4. docker-compose up -d
5. Complete web based initial config

Same results.. which makes me think maybe I'm hitting a bug here...

### What you expected to happen

Would expect the Dashboard page would at least have a placeholder until data is available.  At this point I've added a project with both stats and errors thinking maybe some data or a cron run would clear it up but nothing changed.

Any clues for what to check out would be greatly appreciated!

### PR DESCRIPTION
**Pull request by markstory**
Turn the dashboard features on by default so that dashboards work in self-hosted. This will also apply to other non-saas environments.

Update tests to reflect new default feature state, and add a test to check that updating a dashboard is not possible without the correct feature flags.

### CODE REVIEW

File: tests/sentry/api/endpoints/test_organization_dashboards.py [hunk 1]
```diff
@@ -69,10 +69,13 @@ def test_post_member_can_create(self):
         assert response.status_code == 201
 
     def test_post_features_required(self):
-        response = self.do_request(
-            "post", self.url, data={"title": "Dashboard from Post"}, features=[]
-        )
-        assert response.status_code == 404
+        with self.feature({"organizations:dashbaords-basic": False}):
```
Comment (dashed) [inline tests/sentry/api/endpoints/test_organization_dashboards.py:72]:
```suggestion
        with self.feature({"organizations:dashboards-basic": False}):
```

**Comment by markstory** [comment]
@BYK We can't remove the feature flags. Removing a flag entirely means it cannot be turned off anymore. The dashboard features are restricted by plan in sentry.io, so we'll need to keep the feature flags as long as access is gated by plan.

**Review by dashed — APPROVED** [review]

**Comment by BYK** [comment]
@markstory
>We can't remove the feature flags. Removing a flag entirely means it cannot be turned off anymore. The dashboard features are restricted by plan in sentry.io, so we'll need to keep the feature flags as long as access is gated by plan.

That's my point, if we need them we should make sure they are disabled by default over at getsentry/getsentry before landing this PR.

**Comment by markstory** [comment]
> That's my point, if we need them we should make sure they are disabled by default over at getsentry/getsentry before landing this PR.

I checked getsentry, and the `SubscriptionPlanFeatureHandler` always gives a True/False answer for features which will disable dashboards for accounts that don't have it in their plan's feature list.

**Pull request merged**